pkg/operator/customroute/custom_route_conditions.go

```diff
@@ -0,0 +1,204 @@
+				LastTransitionTime: metav1.Now(),
+				Type:               conditionType,
+				Status:             metav1.ConditionFalse,
+				Reason:             "AsExpected",
+				Message:            "All is well",
+			})
+		}
+	}
+	return conditions
+}
+
+func findCondition(conditions []metav1.Condition, conditionType string) *metav1.Condition {
+	for i := range conditions {
+		if conditions[i].Type == conditionType {
+			return &conditions[i]
+		}
+	}
+	return nil
+}
+
+func checkErrorsConfiguringCustomRoute(errors []error) []metav1.Condition {
+	if len(errors) != 0 {
+		now := metav1.Now()
+		return []metav1.Condition{
+			{
+				LastTransitionTime: now,
+				Type:               "Degraded",
+				Status:             metav1.ConditionTrue,
+				Reason:             "CustomRouteError",
+				Message:            fmt.Sprintf("Error Configuring custom route: %v", errors),
+			},
+			{
+				LastTransitionTime: now,
+				Type:               "Progressing",
+				Status:             metav1.ConditionFalse,
+				Reason:             "CustomRouteError",
+				Message:            fmt.Sprintf("Error Configuring custom route: %v", errors),
+			},
+		}
+	}
+	return nil
+}
+
+func checkIngressURI(ingressConfig *configv1.Ingress, route *routev1.Route) []metav1.Condition {
+	if _, _, err := routeapihelpers.IngressURI(route, route.Spec.Host); err != nil {
+		condition := &metav1.Condition{
+			LastTransitionTime: metav1.Now(),
+			Type:               "Progressing",
+			Status:             "True",
+			Reason:             "RouteNotAdmitted",
+			Message:            fmt.Sprintf("Route not admitted: %v", err),
+		}
+		componentRoute := routecomponenthelpers.GetComponentRouteStatus(ingressConfig, "openshift-authentication", "oauth-openshift")
+		if componentRoute != nil {
+			degradeIfTimeElapsed(componentRoute.Conditions, condition, time.Minute*5)
+		}
+		return []metav1.Condition{*condition}
+	}
+	return nil
+}
+
+// degradeIfTimeElapsed checks if the condition matching this error (same type, reason and message)
+// was found in the set of conditions and its `lastTransitionTime` appeared longer than
+// `maxAge` ago, if so the condition's type is set to "Degraded"
+func degradeIfTimeElapsed(conditions []metav1.Condition, condition *metav1.Condition, maxAge time.Duration) {
+	for i := range conditions {
+		if conditions[i].Reason == condition.Reason &&
+			conditions[i].Message == condition.Message &&
+			conditions[i].Type == condition.Type &&
+			!condition.LastTransitionTime.IsZero() &&
+			condition.LastTransitionTime.Add(maxAge).Before(condition.LastTransitionTime.Time) {
+			condition.Type = "Degraded"
+		}
+	}
+}
+
+func checkRouteAvailablity(secretLister corev1listers.SecretLister, ingressConfig *configv1.Ingress, route *routev1.Route) []metav1.Condition {
+	now := metav1.Now()
+	if err := routeAvailablity(secretLister, route.Spec.Host, ingressConfig); err != nil {
+		condition := &metav1.Condition{
+			LastTransitionTime: now,
```

Comment:
/lgtm cancel

the status updating at https://github.com/openshift/library-go/pull/1232/files/7cfffd862f7a95f225d711d8a4fe78068dd61988#diff-c3fc2dfc23ca84d16aa52c8d592ec59398d92d48d5b8fe6c0d7b9bfe7dec0e50R242 (`updateComponentRouteStatus()`) discards the `LastTransitionTime` but this indeed feels fragile. This was originally written in a fairly more readable manner, I did not notice this change.

@s-urbaniak please, when you're adding changes on top of this branch, do that in separate commits now so that it's easier to review to prevent such things being overlooked as they appear to have been with Sebastian's original patch. We'll eventually squash the commits.

Reply:
I suggested changes, ptal

Reply:
my bad, I thought this got changed but it apparently was not good from the start